src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/contactperson/profiledetails/ContactPersonProfileDetailMigrationService.kt

```diff
@@ -0,0 +1,75 @@
+package uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.contactperson.profiledetails
+
+import kotlinx.coroutines.reactive.awaitSingle
+import org.springframework.data.r2dbc.core.insert
+import org.springframework.data.r2dbc.core.update
+import org.springframework.data.r2dbc.core.R2dbcEntityTemplate
+import org.springframework.data.relational.core.query.Criteria.where
+import org.springframework.data.relational.core.query.Query.query
+import org.springframework.data.relational.core.query.Update.from
+import org.springframework.data.relational.core.sql.SqlIdentifier
+import org.springframework.data.relational.core.sql.SqlIdentifier.unquoted
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+
+@Service
+class ContactPersonProfileDetailMigrationService(
+  private val repository: ContactPersonProfileDetailMigrationMappingRepository,
+  private val template: R2dbcEntityTemplate,
+) {
```

Comment:
There is currently [no support for composite keys](https://github.com/spring-projects/spring-data-relational/issues/574) when using just Spring Data (which we do when using R2DBC) so we use a template instead.